packages/playwright/src/program.ts

```diff
@@ -332,9 +332,39 @@ function resolveReporter(id: string) {
   return require.resolve(id, { paths: [process.cwd()] });
 }
 
+class OptionValidators {
+  static validateNumber(value: string, options: { min?: number, max?: number }): string {
+    if (/[^0-9]/.test(value))
+      throw new InvalidArgumentError('Not a number.');
+    const parsed = parseInt(value, 10);
+    if (isNaN(parsed))
+      throw new InvalidArgumentError('Not a number.');
+    if (options.min !== undefined && parsed < options.min)
+      throw new InvalidArgumentError(`Expected a number greater than ${options.min}.`);
+    if (options.max !== undefined && parsed > options.max)
+      throw new InvalidArgumentError(`Expected a number less than ${options.max}.`);
+    return value;
+  }
+
+  static validateWorkers(value: string): string {
+    if (value[value.length - 1] === '%') {
```

Comment:
If you would like this to be really nice, we can print a better error message, e.g. `Expected a percentage between 1% and 100%`. Similarly in other places.